test/dates/io.jl

```diff
@@ -267,8 +291,7 @@ f = "y m d"
 @test_throws ArgumentError Dates.Date("1 1 32",f)
 @test_throws ArgumentError Dates.Date(" 1 1 32",f)
 @test_throws ArgumentError Dates.Date("# 1 1 32",f)
-# can't find 1st space delimiter,s o fails
-@test_throws ArgumentError Dates.Date("1",f)
+@test Dates.Date("1",f) == Dates.Date(1)
```

Comment:
We have some options if people want this functionality in 0.5. The simplest would be to backport the dispatch for custom parsers which would allow us to get the performance improvement for ISO 8601 formatted strings.